tests/test_default_gaussian.py

```diff
@@ -534,6 +534,53 @@ def test_reduced_state(self):
         self.assertAllAlmostEqual(res[0], expected[0], delta=self.tol)
         self.assertAllAlmostEqual(res[1], expected[1], delta=self.tol)
 
+    def test_samples_coherent(self):
+        """Test that sampled values are calculated correctly by comparing with first and second moments"""
+        self.logTestName()
+        n_samples = 1000000
+        dev = qml.device('default.gaussian', wires=1, hbar=hbar)
+
+        # test correct mean and variance for Homodyne P measurement
+        alpha = 0.324-0.59j
+        dev.apply('CoherentState', wires=[0], par=[alpha])
+        samples = dev.sample('P', [0], [], n_samples)
+        mean = samples.mean()
+        std = samples.var()
+
+        self.assertAlmostEqual(mean, alpha.imag*np.sqrt(2*hbar), delta=1000*self.tol)

```

Comment:
I would not make tolerances here multiples of `self.tol`. `self.tol` is always some value that refers to the _machine precision_ of the underlying system and is meant to account for rounding errors. For example, on my machine (which runs Windows), tol usually is something like 0.001 (don't know why that much) while it can be different on another machine, totally changing what this test accepts.

The reasonable thing here would be to calculate the standard deviation for the given number of samples by hand and then using a multiple of this hardcoded here.